src/types/timestamp.rs

```diff
@@ -234,7 +234,7 @@ impl Timestamp {
     /// NOTE: This is a potentially lossy operation. A Timestamp with picoseconds would return a
     /// number of nanoseconds, losing precision. Similarly, a Timestamp with milliseconds would
     /// also return a number of nanoseconds, erroneously gaining precision.
-    fn fractional_seconds_as_nanoseconds(&self) -> Option<u32> {
+    pub fn fractional_seconds_as_nanoseconds(&self) -> Option<u32> {
```

Comment:
This method name is kind of onerous (I'm the one who wrote it!) and it returns `Option<u32>`, so I think we should keep it private. The public methods should be something like `nanoseconds(&self) -> u32` and `milliseconds(&self) -> u32`. They can use this method under the hood.